span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -485,16 +483,16 @@ private void populateUrlParts(Request.Builder requestBuilder) {
 
     String urlStr = requestBuilder.getUrl();
     try {
-      URL url = new URL(urlStr);
-      requestBuilder.setScheme(url.getProtocol());
+      URI uri = new URI(urlStr);
```

Comment:
Can we add a small descriptive comment on the top, as to why we are using URI, instead of URL, so that it's easier to justify this in future?

Reply:
Done